### PR TITLE
Add test_run parameter to the pw jobs + test

### PR DESCRIPTION
**Pull request by tomdemeyere**
### Checklist

- [ ] I have read the [Contributing Guide](https://quantum-accelerators.github.io/quacc/dev/contributing.html). Don't lie! 😉
- [ ] My PR is on a custom branch and is _not_ named `main`.
- [ ] I have used `black`, `isort`, and `ruff` as described in the [style guide](https://quantum-accelerators.github.io/quacc/dev/contributing.html#style).
- [ ] I have added relevant, comprehensive [unit tests](https://quantum-accelerators.github.io/quacc/dev/contributing.html#unit-tests).

### Notes

- Your PR will likely not be merged without proper and thorough tests.
- If you are an external contributor, you will see a comment from [@buildbot-princeton](https://github.com/buildbot-princeton). This is solely for the maintainers.
- When your code is ready for review, ping one of the [active maintainers](https://quantum-accelerators.github.io/quacc/about/contributors.html#active-maintainers).

## Summary of Changes

This should also improve the missing coverage from #1404

### CODE REVIEW

**Comment by buildbot-princeton** [comment]
Can one of the admins verify this patch?

**Comment by Andrew-S-Rosen** [comment]
My oh my, everything seems to be sporadically breaking all at the same time. Now the ASE website is down, which broke the docs build! *sigh*. Please ignore that one. Everything else should pass...

File: src/quacc/recipes/espresso/core.py [hunk 1]
```diff
@@ -80,6 +81,7 @@ def relax_job(
     relax_cell: bool = False,
     parallel_info: dict[str] | None = None,
     copy_files: str | Path | list[str | Path] | None = None,
+    test_run: bool = False,
```
Comment (Andrew-S-Rosen) [inline src/quacc/recipes/espresso/core.py:84]: Please add a docstring for `test_run`  in both places added. Also, while you're at it, if you can have it be above `copy_files` that'd be great (I usually reserve that one for last, before the **kwargs). Just make sure the docstring placement matches.

Comment (tomdemeyere) [reply src/quacc/recipes/espresso/core.py:84]: Done!

Comment (Andrew-S-Rosen) [reply src/quacc/recipes/espresso/core.py:84]: Great! There's a failing Espresso test, btw.

Comment (tomdemeyere) [reply src/quacc/recipes/espresso/core.py:86]: Yes, and quite important: the test_run part wasn't working! It should be better now

**Comment by Andrew-S-Rosen** [comment]
Merging, thanks! Just noticed the grid phonon test is running for 300 s but I can try to chip away at that elsewhere (might not be possible to do much better though).

**Pull request merged**